website/project/views/register.py

```diff
@@ -151,10 +447,42 @@ def node_register_template_page_post(auth, node, **kwargs):
     schema = MetaSchema.find(
         Q('name', 'eq', template)
     ).sort('-schema_version')[0]
+
+    # Create the registration
     register = node.register_node(
         schema, auth, template, json.dumps(clean_data),
     )
 
+    if data['registrationChoice'] == 'Make registration public immediately':
+        register.is_public = True
+        register.save()
+    elif data['registrationChoice'] == 'Enter registration into embargo':
+        # Sanitize embargo end date
+        embargo_end_date = datetime.datetime.strptime(
+            data['embargoEndDate'],
+            "%a, %d %b %Y %H:%M:%S %Z"
+        )
+
+        # Initiate embargo
+        try:
+            register.embargo_registration(auth.user, embargo_end_date)
+            register.add_log(
+                action=NodeLog.EMBARGO_INITIATED,
+                params={
+                    'node': register._id,
+                    'embargo_id': register.embargo._id,
+                },
+                auth=auth,
+                log_date=datetime.datetime.utcnow(),
```

Comment:
I believe that `NodeLog.log_date` defaults to `utcnow` upon creation. Therefore, there is no need to pass this in directly. In fact, allowing this to be set on save will yield ever-so-slightly more accurate timestamps.